src/modes/simple_select.js

```diff
@@ -233,16 +234,82 @@ module.exports = function(ctx, options = {}) {
       }
     },
     render: function(geojson, push) {
-      geojson.properties.active = (ctx.store.isSelected(geojson.properties.id) )
+      geojson.properties.active = (ctx.store.isSelected(geojson.properties.id))
         ? Constants.activeStates.ACTIVE
         : Constants.activeStates.INACTIVE;
       push(geojson);
       if (geojson.properties.active !== Constants.activeStates.ACTIVE
         || geojson.geometry.type === Constants.geojsonTypes.POINT) return;
       createSupplementaryPoints(geojson).forEach(push);
     },
-    trash() {
+    trash: function() {
       ctx.store.delete(ctx.store.getSelectedIds());
+    },
+    combineFeatures: function() {
+      var selectedFeatures = ctx.store.getSelected();
+
+      if(!selectedFeatures || selectedFeatures.length < 2) return;
+
+      var featureType = selectedFeatures[0].type.replace('Multi','');
+      var coordinates = [];
+      var properties = selectedFeatures[0].properties;
+
+      selectedFeatures = selectedFeatures.map(function(feature) {
+        if(feature.type.replace('Multi','') !== featureType) {
+          return;
+        }
+        if(feature.type.includes('Multi')) {
+          feature.getCoordinates().forEach(function(subcoords) {
+            coordinates.push(subcoords);
+          });
+        } else {
+          coordinates.push(feature.getCoordinates());
+        }
+
+        return feature.toGeoJSON();
+      });
+
+      var multiFeature = new MultiFeature(ctx, {
+        type: Constants.geojsonTypes.FEATURE,
+        properties: selectedFeatures[0].properties || {},
+        geometry: {
+          type: 'Multi' + featureType,
+          coordinates: coordinates
+        }
+      });
+      ctx.store.add(multiFeature);
+      ctx.store.delete(ctx.store.getSelectedIds(), { silent: true });
+      ctx.store.setSelected([multiFeature.id]);
+
+      ctx.map.fire(Constants.events.MERGE, {
+        createdFeatures: [multiFeature.toGeoJSON()],
+        deletedFeatures: selectedFeatures
+      });
+    },
+    uncombineFeatures: function() {
+      var selectedFeatures = ctx.store.getSelected();
+      if (!selectedFeatures) return;
+
+      var createdFeatures = [];
+
+      selectedFeatures = selectedFeatures.map(function(feature){
+        if(feature instanceof MultiFeature) {
+          feature.getFeatures().forEach(function(subFeature){
+            ctx.store.add(subFeature);
+            subFeature.properties = feature.properties || {};
+            createdFeatures.push(subFeature.toGeoJSON());
+            ctx.store.select([subFeature.id]);
+          });
+          ctx.store.delete(feature.id, { silent: true });
+        }
+
+        return feature.toGeoJSON();
+      })
+
+      ctx.map.fire(Constants.events.UNMERGE, {
+        createdFeatures: createdFeatures,
+        deletedFeatures: selectedFeatures
```

Comment:
This is saying all features selected were deleted. Line 303 says that only features that are a MultiFeature get deleted.